src/trans.rs

```diff
@@ -1034,7 +1091,7 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
 
     #[inline]
     fn type_size(&self, ty: Ty<'tcx>) -> usize {
-        let substs = self.tcx.mk_substs(Substs::empty());
+        let substs = Substs::empty(*self.tcx);
```

Comment:
Nope, but `Substs::empty` takes a struct, not a reference.  FWIW, that seems like a common pattern with `TyCtxt`: lots of functions ownership of it, but the `TyCtxt` has `#[derive(Clone, Copy)]` on it.  Just glancing at it, I hope the compiler is doing some magic to eliminate those copies, because `TyCtxt` is a pretty big struct.